src/Server.ts

```diff
@@ -48,12 +49,16 @@ export class Server {
     // "presence" option is not used from now on
     delete options.presence;
 
-    registerGracefulShutdown((signal) => {
-      this.matchMaker.gracefullyShutdown().
-        then(() => this.shutdown()).
-        catch((err) => debugError(`error during shutdown: ${err}`)).
-        then(() => process.exit());
-    });
+    if(options.shutdownProcessOnSignal !== false) {
```

Comment:
I think that just checking for truthiness looks a bit clean

```suggestion
    if(options.shutdownProcessOnSignal) {
```

Reply:
The `!== false` keeps this from breaking existing behavior since users will need to pass `shutdownProcessOnSignal: false` explicitly to disable the shutdown process. Without it, everyone who updates will need to pass `shutdownProcessOnSignal: true` to keep the existing behavior.

Reply:
So this is kinda weird, because basically you're relying on the fact that only `false` is the disabling value, whereas `undefined` and `null` are values that enable the behaviour.

Considering the name of the flag, and the fact that it's enabled by default, there should be a place which sets a default value of `true` to the flag (maybe in the wrapping function parameter destructuring), and this will mean that by default, you need to check for a truthy (`!!`) value to turn on the behaviour.

Reply:
Yeah that is a bit strange. I changed it to use a default value and destructuring:

```
const { shutdownProcessOnSignal = true } = options;
if(shutdownProcessOnSignal) {
```

Reply:
Looks better 👍 gj